src/d3a/d3a_core/sim_results/kpi.py

```diff
@@ -159,10 +159,24 @@ def area_performance_indices(self, area):
                                self.state[area.name].total_energy_produced_wh
         return {"self_sufficiency": self_sufficiency, "self_consumption": self_consumption}
 
+    def _kpi_ratio_to_percentage(self, area_name):
+        if self.performance_indices[area_name]["self_sufficiency"] is not None:
+            self_sufficiency_percentage = \
+                self.performance_indices[area_name]["self_sufficiency"] * 100
+        else:
+            self_sufficiency_percentage = 0.0
+        if self.performance_indices[area_name]["self_consumption"] is not None:
+            self_consumption_percentage = \
+                self.performance_indices[area_name]["self_consumption"] * 100
+        else:
+            self_consumption_percentage = 0.0
+        return {"self_sufficiency": self_sufficiency_percentage,
```

Comment:
No need to return the variables here and reassign, you already change the `self.performance_indices` member of the class.

Reply:
Sorry, I couldn't follow you, I only have used `self.performance_indices`  for calculation and assigned the calculated value to a local variable called `self_consumption_percentage & self_sufficiency_percentage `

Reply:
I see now why you need a return variable here, you need to only update the stats sent to redis and not the normal kpi stats. That's good , LGTM once more :P